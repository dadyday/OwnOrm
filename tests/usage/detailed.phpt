<?php
require_once 'cfg.php';

use OwnOrm\Model;
use Tester\Assert as Is;

// detailed config usage w/o chain and magic

$oModel = new Model([
	'user' => [
		'properties' => [
			'name' => ['type' => 'string', 'length' => 30, 'default' => 'guest', 'null' => false],
			'password' => ['type' => 'string', 'length' => 40, 'default' => null, 'null' => true],
		],
		'indices' => [
			'name' => ['type' => 'unique', 'fields' => ['name']],
		],
	],
	'login' => [
		'properties' => [
			'user_id' => ['type' => 'integer', 'related' => 'user'],
			'logged' => ['type' => 'date', 'default' => 'now'],
		],
		'indices' => [
			'userlogin' => ['type' => 'index', 'fields' => ['user_id', 'logged']],
		],
	]
]);

include 'check.php';
