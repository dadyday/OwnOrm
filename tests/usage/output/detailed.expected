OwnOrm\Entity::__set_state(/* #5bc6 */ array(
	'oModel' => OwnOrm\Model::__set_state(/* #6792 */ array(
		'aEntity' => array(
			'user' => /* OwnOrm\Entity dumped on line 1 */,
			'login' => OwnOrm\Entity::__set_state(/* #7bc1 */ array(
				'oModel' => /* OwnOrm\Model dumped on line 2 */,
				'name' => 'login',
				'aProperty' => array(
					'user_id' => OwnOrm\Property::__set_state(/* #aede */ array(
						'oEntity' => /* OwnOrm\Entity dumped on line 5 */,
						'name' => 'user_id',
						'null' => TRUE,
						'type' => NULL,
						'length' => 0,
						'default' => NULL,
						'related' => 'user',
					)),
					'logged' => OwnOrm\Property::__set_state(/* #6d01 */ array(
						'oEntity' => /* OwnOrm\Entity dumped on line 5 */,
						'name' => 'logged',
						'null' => TRUE,
						'type' => 'string',
						'length' => 40,
						'default' => 'now',
						'related' => NULL,
					)),
				),
				'aIndex' => array(
					'userlogin' => OwnOrm\Index::__set_state(/* #ff0a */ array(
						'oEntity' => /* OwnOrm\Entity dumped on line 5 */,
						'name' => 'userlogin',
						'type' => 'index',
						'aField' => array(
							'user_id' => /* OwnOrm\Property dumped on line 9 */,
							'logged' => /* OwnOrm\Property dumped on line 18 */,
						),
					)),
				),
			)),
		),
		'aRow' => array(),
	)),
	'name' => 'user',
	'aProperty' => array(
		'name' => OwnOrm\Property::__set_state(/* #c2d6 */ array(
			'oEntity' => /* OwnOrm\Entity dumped on line 1 */,
			'name' => 'name',
			'null' => FALSE,
			'type' => 'string',
			'length' => 30,
			'default' => 'guest',
			'related' => NULL,
		)),
		'password' => OwnOrm\Property::__set_state(/* #7e5a */ array(
			'oEntity' => /* OwnOrm\Entity dumped on line 1 */,
			'name' => 'password',
			'null' => TRUE,
			'type' => 'string',
			'length' => 40,
			'default' => NULL,
			'related' => NULL,
		)),
	),
	'aIndex' => array(
		'name' => OwnOrm\Index::__set_state(/* #3a96 */ array(
			'oEntity' => /* OwnOrm\Entity dumped on line 1 */,
			'name' => 'name',
			'type' => 'unique',
			'aField' => array('name' => /* OwnOrm\Property dumped on line 45 */),
		)),
	),
))
